```{r standalone setup, echo=FALSE}
#strain<-"Trp53cor1"
#timepoint<-"Embryonic"
#filename<-"Leg_WT_v_KO"
#dir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/Trp53cor1/"
### THINGS TO REMEMBER: change oldbam to bam and olddiff to diff once new set done! 
```

`r strain` KO vs WT (`r timepoint`)
======================================

```{r init, echo=FALSE}
# Set knitr opts

#path<-paste(filename,"/figure/",sep="")
opts_chunk$set(echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=4,dev=c('png', 'pdf'))
alpha<-0.05 

analysisdir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/"
#diffdir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/"
GTF <- "/n/rinn_data1/seq/lgoff/Projects/BrainMap/data/annotation/mm10_gencode_vM2_with_lncRNAs_and_LacZ.gtf"
genome<-"mm10"

```

```{r vars_and_setup,echo=FALSE,results='hide',message=FALSE}
library(cummeRbund)
library(xtable)


#setwd(analysisdir)
deletionCoords<-read.table("/n/rinn_data1/users/agroff/GITHUB/BrainMap/analysis/mm10DeletionCoords.txt",sep="\t",header=TRUE,stringsAsFactors=FALSE)
colnames(deletionCoords)<-c("gene_name","gene_region","deletionRegion")
```


```{r cummeRbund_and_data_setup, echo=FALSE}
cuff<-readCufflinks(dir=dir,gtfFile="/n/rinn_data1/seq/lgoff/Projects/BrainMap/data/annotation/mm10_gencode_vM2_with_lncRNAs_and_LacZ.gtf",genome=genome)
```



# P21, lincp21, lacZ barplots


```{r barplots,fig.height=8, fig.width=8}
genotypingGeneIDs<-c(strain,"LacZ","Cdkn1a")
genotypingGenes<-getGenes(cuff,genotypingGeneIDs)
expressionBarplot(genotypingGenes,replicates=T)
```

## Session Info
```{r session}
sessionInfo()
```

## Run Info
```{r runInfo}
runInfo(cuff)
```

