```{r test, echo=FALSE}
#dir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/Trp53cor1_E14.5_Lung_b0329/"
#dir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/Trp53cor1_E14.5_Leg_b0329/"
#dir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/oldDiffs/Trp53cor1_E14.5_HEART/"
dir<-"/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/Trp53cor1_E14.5_Liver_b0329/"
tissue<-"Leg"
```

Figure 3 template 
========================================================

```{r tissue, echo=FALSE}
dir<-dir
tissue<-tissue
```

```{r opts, echo=FALSE}
opts_chunk$set(echo=FALSE, message=FALSE, warning=FALSE,fig.height=4, fig.width=4,dev=c('png', 'pdf'))
```


```{r setup, echo=FALSE, results='hide'}
library(cummeRbund)
library(limma)
library(GSA)
library(gplots)
library(marray)
library(ggplot2)
library(gtable)
library(gridExtra)
library(RColorBrewer)
library(RMySQL)

alpha=0.05

cuff<-readCufflinks(dir=dir,gtfFile="/n/rinn_data1/seq/lgoff/Projects/BrainMap/data/annotation/mm10_gencode_vM2_with_lncRNAs_and_LacZ.gtf",genome="mm10")

all_gs<-GSA.read.gmt("/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/diff/p53_genesets/msigdb.v4.0.symbols.gmt")
```


Figure 3 plots for `r tissue`

```{r lincp21, fig.width=7, fig.height=6}
lincp21<-getGene(cuff,"Trp53cor1")
lacZ<-getGene(cuff,"LacZ")
expressionBarplot(lincp21,replicates=TRUE)+ylim(0,2)+theme_bw()
```

```{r lacZ, fig.width=7, fig.height=6}
expressionBarplot(lacZ,replicates=TRUE)+ylim(0,7)+theme_bw()
```

```{r DEgenes}
DEids<-getSig(cuff,alpha=0.05)
DEgenes<-getGenes(cuff,DEids)
```

Number of DE genes (FDR 0.05): `r length(DEids)`

```{r DEgenes_FC_Heatmap, fig.width=7, fig.height=6}
csFoldChangeHeatmap(DEgenes,"WT",replicates=TRUE,replicate_num=0,cluster="row")
```

Universe DE genes foldchange heatmap:
```{r UNIVERSE_DEgenes_FC_Heatmap, fig.width=7, fig.height=6}
universe_geneIDs<-read.table("/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/analysis/universe_of_DE_genes.tab",stringsAsFactors=FALSE)
universe_geneIDs<-universe_geneIDs$x
DEgenes<-getGenes(cuff,universe_geneIDs)
csFoldChangeHeatmap(DEgenes,"WT",replicates=TRUE,replicate_num=0,cluster="row")
```



```{r expressedandchanging_setup,results='hide'}
#alternately, anything expressed and over logfoldchange2
dat<-diffData(genes(cuff))
dat<-dat[which(abs(dat$log2_fold_change)>2),]
dat<-dat[which(dat$value_1>0.5|dat$value_2>0.5),]
ids<-dat$gene_id
genes<-getGenes(cuff,ids)
```

Now, look at genes that are detectably expressed (FPKM>0.5, yes I know it should be 1), and log2foldchange>2 in any direction: `r length(ids)` genes

```{r expressed_and_changing, eval=FALSE, fig.width=7, fig.height=6}
csFoldChangeHeatmap(genes,"WT",replicates=TRUE,replicate_num=0,cluster="both")
```

```{r expressedandchanging_sig_setup, eval=FALSE}
sig<-dat[which(dat$significant=="yes"),]
ids<-sig$gene_id
genes<-getGenes(cuff,ids)
#Now, those that are expressed changing and called sig: `r length(ids)` genes 
```




```{r expressed_and_changing_andSIG, eval=FALSE, fig.width=7, fig.height=6}
csFoldChangeHeatmap(genes,"WT",replicates=TRUE,replicate_num=0,cluster="both")
```



Tracks for Glo1: 

```{r track_vis, fig.height=15, fig.width=10}
name<-"Glo1"
name<-"Cdkn1a"
myGene<-getGene(cuff,name)

library(GenomicFeatures)
library(Gviz)

real_chromInfo<-read.table("/n/rinn_data1/users/agroff/GITHUB/BrainMap/abbie_annotation/real_chromosomes_mm10_brainmap.chrom.info",header=TRUE)
genome<-"mm10"

mm10DB<-loadDb("/n/rinn_data1/users/agroff/GITHUB/BrainMap/analysis/mm10gencode_brainmapDB_nolacz.sqlite")

#Helper Functions
movingAverage <- function(x, n=10, centered=TRUE) {
  
  if (centered) {
    before <- floor  ((n-1)/2)
    after  <- ceiling((n-1)/2)
  } else {
    before <- n-1
    after  <- 0
  }
  
  # Track the sum and count of number of non-NA items
  s     <- rep(0, length(x))
  count <- rep(0, length(x))
  
  # Add the centered data
  new <- x
  # Add to count list wherever there isn't a
  count <- count + !is.na(new)
  # Now replace NA_s with 0_s and add to total
  new[is.na(new)] <- 0
  s <- s + new
  
  # Add the data from before
  i <- 1
  while (i <= before) {
    # This is the vector with offset values to add
    new   <- c(rep(NA, i), x[1:(length(x)-i)])
    
    count <- count + !is.na(new)
    new[is.na(new)] <- 0
    s <- s + new
    
    i <- i+1
  }
  
  # Add the data from after
  i <- 1
  while (i <= after) {
    # This is the vector with offset values to add
    new   <- c(x[(i+1):length(x)], rep(NA, i))
    
    count <- count + !is.na(new)
    new[is.na(new)] <- 0
    s <- s + new
    
    i <- i+1
  }
  
  # return sum divided by count
  s/count
}

makeBamTrack<-function(bamFile,bamName,genome=genome,chromosome,color="steelblue",window=20,ylim=c(0,300)){
  
track<-DataTrack(range=bamFile,name=bamName,genome=genome,type="h",transformation=function(x){movingAverage(x,window)},col=color,fill.histogram=color,col.histogram=color,chromosome=chromosome, ylim=ylim, lwd=1.5)
  
  return(track)
}

# Constants #########
annot<-annotation(myGene)
margin<-1000
locus<-strsplit(annot$locus,":")
locus<-unlist(locus)
chrom<-locus[[1]]
start_and_end<-strsplit(locus[[2]],"-")
start_and_end<-unlist(start_and_end)
from<-as.numeric(start_and_end[[1]])-margin
to<-as.numeric(start_and_end[[2]])+margin

genetrack<-GeneRegionTrack(mm10DB,rstarts=from,rends=to,chromosome=chrom,showId=TRUE,geneSymbol=TRUE,genome=genome,name="LincRNA Isoforms",fill="steelblue")

reps<-replicates(cuff)
files<-lapply(reps$file,function(x){strsplit(x, "/")})
files<-as.data.frame(files)
samples<-(t(files[11,]))
rownames(samples)<-NULL
JRs<-samples


bamRoot<-'/n/rinn_data1/users/agroff/seq/OtherMice/Diana/lincP21/bam/'

bamFiles<-lapply(JRs,function(x){paste(bamRoot,x,"/accepted_hits.bam",sep="")})

bamNames<-reps$rep_name

specCols<-brewer.pal(3,"Paired")
colPal<-colorRampPalette(specCols)
bamColors<-colPal(length(bamFiles))


doPlot<-function(genome=genome,name,myChr,from,to,window,bamFiles,bamNames){
  #Make Tracks
  axTrack<-GenomeAxisTrack(add53 = TRUE,add35 = TRUE, labelPos = "above")
  idxTrack <- IdeogramTrack(genome = genome, chromosome = myChr)
  #BamTracks
  write("\tBamTracks",stderr())
  bamTracks<-list()
  bamOrder<-c(1:length(bamFiles))

  for (i in bamOrder){
    track<-makeBamTrack(bamFiles[[i]],bamNames[[i]],genome=genome,chromosome=myChr,color=bamColors[i],window=window)
    bamTracks<-c(bamTracks,track)
  }
  
  #Plot Tracks
  write("\tplotting...",stderr())
  # myTracks<-c(bamTracks,knownGenes)
  myTracks<-c(idxTrack,axTrack,genetrack,bamTracks)
  trackSizes<-c(1,1,4,rep(1,length(bamTracks)))

  plotTracks(myTracks,from=from,to=to,chromosome=myChr,showAxis=FALSE,background.title="black",col.title="white",col.axis="black",sizes=trackSizes,geneSymbol=TRUE)
}

pdf("Cdkn1a_liver_tracks.pdf")

doPlot(genome=genome, name=name, myChr=chrom, from=from, to=to, window=20,bamFiles=bamFiles, bamNames=bamNames)
dev.off()
detach("package:GenomicFeatures")

```




GO terms associated with sigDE genes: 

```{r DEgenes_GO, fig.width=7, fig.height=6, eval=FALSE}
library(ReactomePA)
library(DOSE)
geneAnnot<-annotation(DEgenes)
geneNames<-geneAnnot$gene_short_name
library(clusterProfiler)
require(biomaRt)

ensembl <- useMart("ENSEMBL_MART_ENSEMBL","mmusculus_gene_ensembl", 
host="www.ensembl.org")

#Get entrezIDs
getEntrezIDs<-function (geneNames)
{
    tmp <- getBM(attributes = c("entrezgene"), filters = "mgi_symbol", values = geneNames, mart = ensembl)
    tmp
}
sigEntrez<-getEntrezIDs(geneNames)
sigEZ<-strsplit(as.character(sigEntrez), ", ")
sigEZ<-unlist(sigEZ)

require(org.Mm.eg.db) #mapping of entrez IDs to genbank

goBP<-enrichGO(gene=sigEZ, organism="mouse",ont="BP",pvalueCutoff=0.01,readable=T)

goMF<-enrichGO(gene=sigEZ, organism="mouse",ont="MF",pvalueCutoff=0.01,readable=T)

goCC<-enrichGO(gene=sigEZ, organism="mouse",ont="CC",pvalueCutoff=0.01,readable=T)

kegg<-enrichKEGG(gene=sigEZ, organism="mouse",pvalueCutoff=.01, readable=T)

pathway<-enrichPathway(gene=sigEZ,organism="mouse",pvalueCutoff=0.01, readable=T)

```


```{r GO_images, fig.width=8, fig.heigh=8}
plot(goBP,showCategory=20) + theme(axis.text.x=element_text(angle=-90,hjust=0)) + ggtitle("Enriched BP")+theme_bw()

plot(goMF,showCategory=20) + theme(axis.text.x=element_text(angle=-90,hjust=0)) + ggtitle("Enriched MF")+theme_bw()

plot(goCC,showCategory=20) + theme(axis.text.x=element_text(angle=-90,hjust=0)) + ggtitle("Enriched CC")+theme_bw()

plot(kegg, showCategory=20) + ggtitle("Kegg Pathways")+theme_bw()

plot(pathway,showCategory=20) + ggtitle("Reactome pathway enrichment")+theme_bw()

detach("package:biomaRt")
```


```{r gsea_functions, eval=FALSE}

gene_set_index <- function(genelist, short_names){
  which(short_names %in% genelist)   
}

get_gene_set_p_vals <- function(input, gs, alternative){
  gene_set_indices <- lapply(gs$genesets, function(genelist){
    gene_set_index(genelist,input$short_name)
  })
  pvl<-lapply(gene_set_indices,geneSetTest,input$test_stat, alternative=alternative)
  pvl_mat<-as.data.frame(t(unlist(pvl)))
  colnames(pvl_mat) <- gs$geneset.names
  return(pvl_mat)
}

get_gene_set_q_vals <- function(pvl_mat, method="bonferroni"){
  comp_corrected <- matrix(p.adjust(pvl_mat, method=method), nrow=nrow(pvl_mat), ncol=ncol(pvl_mat))
  colnames(comp_corrected) <- colnames(pvl_mat)
  rownames(comp_corrected) <- rownames(pvl_mat)
  return(comp_corrected)
}

colMins<-function(x){
  apply(x,2,min)
}
rowMins<-function(x){
  apply(x,1,min)
}

ztest<-function(samp,pop){
  (mean(samp,na.rm=T)-mean(pop,na.rm=T))/sd(pop,na.rm=T)
}


get_gene_set_ztest <- function(scoring_df, gs){  
  gene_set_indices <- lapply(gs$genesets, function(genelist){
    gene_set_index(genelist, scoring_df$short_name)
  })
  zscores <- lapply(gene_set_indices,function(gsi){
    ztest(scoring_df$test_stat[gsi],scoring_df$test_stat)
  })
  
  zscore_mat<-do.call(rbind,lapply(zscores,unlist))
  rownames(zscore_mat) <- gs$geneset.names
  colnames(zscore_mat) <- "zscore"
  return(zscore_mat)
}
  
```


```{r GSEA_all, fig.width=7, fig.height=6, eval=FALSE}
population<-genes(cuff)
population.diff<-diffData(population)
annotation<-annotation(genes(cuff))
gene_names<-merge(annotation,population.diff)

InputCols<-maPalette(low="white",high="red",k=100)
df.pop<-data.frame("short_name"=toupper(gene_names$gene_short_name),"test_stat"=gene_names$test_stat)
df.pop.unique<-unique(df.pop)
rownames(df.pop.unique)<-NULL
df.pop.unique.ordered<-df.pop.unique[order(df.pop.unique$test_stat),]
Input.df<-df.pop.unique.ordered
Input.df$short_name<-as.character(Input.df$short_name)

ALL_pval_mat<-get_gene_set_p_vals(Input.df,all_gs,alternative="either")
all_pval_corrected<-get_gene_set_q_vals(ALL_pval_mat)
all_pval_corrected<-rbind(all_pval_corrected,all_pval_corrected)

all_zscores<-get_gene_set_ztest(Input.df,all_gs)
all_zscores<-cbind(all_zscores,all_zscores)

k <- 100
myColors<-maPalette(low="blue",mid="white",high="red",k=k)
myBreaks<-seq(-2,2,length.out=(k+1))
enrichmentBreaks<-seq(0,6,length.out=(k+1))

x<-(-log10(t(all_pval_corrected[,which(colMins(all_pval_corrected) < 0.0001)])))
x_ordered<-x[order(x[,1], decreasing=TRUE),]
gseaEnriched<-length(x_ordered)
#There are `r gseaEnriched` categories enriched (p<0.0001). They are: 

```



```{r gsea_enrich_fig, fig.width=8, fig.height=16, eval=FALSE}
if(length(x_ordered)>500){x_ordered<-x_ordered[1:500,]}
noinfinities_x<-x_ordered[which(x_ordered !="Inf")]
x_max<-max(noinfinities_x)+100
x_ordered[x_ordered=="Inf"]<-x_max


if(dim(x_ordered)[1]>1){
  heatmap.2(x_ordered, trace="none",col=InputCols,breaks=enrichmentBreaks, margins=c(1,20),dendrogram="both",labCol=c(""),cexRow =1, offsetRow=0)
}else{print("Not enough significant categories to print a heatmap!")}

rm(x_ordered)
rm(x_max)
rm(x)
```


Zscore:

```{r gsea_zscore_fig, fig.width=8, fig.height=16, eval=FALSE}
#ALL zscore: 
x<-all_zscores[which(colMins(all_pval_corrected) < 0.0001),]
x_ordered<-x[order(x[,1], decreasing=TRUE),]

if(length(x_ordered)>500){x_ordered<-x_ordered[1:500,]}
x_ordered<-as.matrix(x_ordered)

if(dim(x_ordered)[1]>1){
  heatmap.2(x_ordered, trace="none",col=myColors,breaks=myBreaks,margins=c(1,20),dendrogram="both",labCol=c(""),cexRow = 1, offsetRow=0)
}else{print("Not enough significant categories to print a heatmap!")}

rm(x_ordered)
rm(x)
```
